src/test/ui/generator/unresolved_type_param.rs

```diff
@@ -0,0 +1,9 @@
+// edition:2018
+#![feature(futures_api, async_await, await_macro)]
+async fn bar<T>() -> () {}
+
+async fn foo() {
+        await!(bar());
```

Comment:
Similarly, it would be good to include a `//~ ERROR` tag for "cannot infer type for `T`" here, since the purpose of this test is basically to check that indeed we *do* emit this diagnostic. (That is, include `ERROR` labels for things where, if they disappeared, the test may not be serving its original purpose.)

Reply:
Really good spot, I hadn't realised I could use NOTE. Fixed.